fork_choice/confirmation-rule.md

```diff
@@ -0,0 +1,431 @@
+
+This rule makes the following network synchrony assumption: starting from the current slot, attestations created by honest validators in any slot are received by the end of that slot.
+Consequently, this rule provides confirmations to users who believe in the above assumption. If this assumption is broken, confirmed blocks can be reorged without any adversarial behavior, and without slashing.
+
+There are two algorithms in the document:
+- [**Confirmation Rule**](#confirmation-rule): Given a block and confirmation safety parameters, outputs whether the block is confirmed.
+- [**Confirmation Score**](#confirmation-score): Given a block, outputs the confirmation score for the block, i.e., the maximum possible confirmation safety parameters to deem the block confirmed.
+
+*Note:* These algorithms use floating point arithmetic in some places. The rest of `consensus-specs` uses `uint64` arithmetic exclusively to ensure that Ethereum clients arrive at the exact same results - a property crucial for consensus objects (such as the `BeaconState`). This document describes a local confirmation rule that is not used to derive any consensus objects. Using floating point arithmetic here allows for a more readable specification, and client-native floating point arithmetic provides sufficient precision for its objective.
+
+## Confirmation Rule
+
+This section specifies an algorithm to determine whether a block is confirmed. The confirmation rule can be configured to the desired tolerance of Byzantine validators, for which the algorithm takes the following input parameters:
+| Input Parameter                    | Type     | Max. Value                         | Description                                                            |
+| ---------------------------------- | -------- |:---------------------------------- | ---------------------------------------------------------------------- |
+| `confirmation_byzantine_threshold` | `uint64` | `33`                               | the maximum percentage of Byzantine validators among the validator set |
+| `confirmation_slashing_threshold`  | `uint64` | `confirmation_byzantine_threshold` | the maximum percentage of slashings among the validator set            |
+
+
+### Helper Functions
+
+#### `get_committee_weight_between_slots`
+
+```python
+def get_committee_weight_between_slots(store: Store, start_slot: Slot, end_slot: Slot) -> Gwei:
+    """Returns the total weight of committees between ``start_slot`` and ``end_slot`` (inclusive of both).
+    Uses the justified state to compute committee weights.
+    """
+    justified_state = store.checkpoint_states[store.justified_checkpoint]
+    total_active_balance = get_total_active_balance(justified_state)
+
+    # If an entire epoch is covered by the range, return the total active balance
+    start_epoch = compute_epoch_at_slot(start_slot)
+    end_epoch = compute_epoch_at_slot(end_slot)
+    if end_epoch > start_epoch + 1:
+        return total_active_balance
+
```

Comment:
```suggestion

 committee_weight = total_active_balance // SLOTS_PER_EPOCH
```

Reply:
`committee_weight` should be moved here (see other suggested change [here](https://github.com/ethereum/consensus-specs/pull/3339/files#r1219513367))